include/seqan3/search/algorithm/detail/search_trivial.hpp

```diff
@@ -66,15 +77,20 @@ inline bool search_trivial(cursor_t cur, query_t & query, typename cursor_t::siz
     else
     {
         // Insertion
-        if (error_left.insertion > 0)
+        // Only allow insertions if there is no match and we are not at the beginning of the query.
+        bool const allow_insertion = cur.query_length() > 0 ? cur.last_rank() != to_rank(query[query_pos]) : true;
+
+        if (allow_insertion && (prev_error != error_type::DELETION || error_left.substitution == 0) &&
+            error_left.insertion > 0)
         {
             search_param error_left2{error_left};
             error_left2.insertion--;
             error_left2.total--;
 
-            // always perform a recursive call. Abort recursion if and only if recursive call found a hit and
+            // Always perform a recursive call. Abort recursion if and only if recursive call found a hit and
             // abort_on_hit is set to true.
-            if (search_trivial<abort_on_hit>(cur, query, query_pos + 1, error_left2, delegate) && abort_on_hit)
+            if (search_trivial<abort_on_hit>(cur, query, query_pos + 1, error_left2, error_type::INSERTION, delegate)
+                && abort_on_hit)
```

Comment:
you already wrap the condition expression below. So please to follow the style guide, put every argument of the function on a new line and then put the && behind the closing body. Also please add `{}` after that so it becomes easier to read when the expression ends an when the block starts.

Reply:
I also changed function definition after reading the style guide.